lib/info-extras.js

```diff
@@ -124,11 +124,49 @@ exports.getPublished = (body) => {
  * @return {Array.<Object>}
  */
 exports.getRelatedVideos = (body) => {
-  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': {"rvs":', '},');
+  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': ', ',\n');
+  let watchNextJson;
   try {
     jsonStr = JSON.parse(jsonStr);
-  } catch (err) {
+    watchNextJson = JSON.parse(jsonStr.watch_next_response);
+  }
+  catch (err) {
     return [];
   }
-  return jsonStr.split(',').map((link) => qs.parse(link));
-};
+  let rvsParams = jsonStr.rvs.split(',').map((e) => {
+    return {
+      id: util.between(e, 'id=', '&'),
+      shortViewCount: util.between(e, 'short_view_count_text=', '+views')
+    };
+  });
+  let secondaryResults = watchNextJson.contents.twoColumnWatchNextResults.secondaryResults.secondaryResults.results;
+  let recomendedVideos = [];
+  for (let result of secondaryResults) {
+    let details;
+    if (details = result.compactVideoRenderer) {
+      try {
+        let viewCount = details.viewCountText.simpleText;
+        if(!viewCount)
+          viewCount = details.viewCountText.runs[0].text.replace(' watching', '');
+        let shortViewCount = details.shortViewCountText.simpleText;
+        if(!shortViewCount.match(/^\d/) || !shortViewCount)
+        {
+          if(shortViewCount = rvsParams.find((elem) => elem.id === details.videoId))
+            shortViewCount = shortViewCount.shortViewCount;
+        }
+        recomendedVideos.push({
+          id: details.videoId,
+          title: details.title.simpleText,
+          author: details.shortBylineText.runs[0].text,
+          ucid: details.shortBylineText.runs[0].navigationEndpoint.browseEndpoint.browseId,
+          author_thumbnail: details.channelThumbnail.thumbnails[0].url,
+          short_view_count_text: shortViewCount.replace(' views', ''),
+          view_count: (viewCount.match(/^\d/)) ? viewCount.replace(' views','') : shortViewCount.replace(' views',''),
```

Comment:
no need for parens around the if statement. do the same to `.split()` it instead of replacing " views"